### DISCUSSION TITLE
[14일차-지원] 제너레이터에 대해 설명해주세요

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#47] [프론트 면접 질문]
제너레이터에 대해 설명해주세요

**Comment by ez1n** [comment]
**Generator**
중간에 원하는 부분에서 멈추었다가, 그 부분부터 다시 실행할 수 있는 능력을 가진 함수 
⇒ Iterator를 반환
⇒ ES6에서 도입되어 함수 선언식의 함수명 앞에 별(*)표를 붙여 사용
⇒ next() 메서드를 호출할 때마다 반환할 값은 yield 키워드를 통해 지정

**Iterator**
반복을 위해 설계된 특정 인터페이스가 있는 객체
⇒ 결과 객체를 반환하는 next() 메서드를 가지고 있음
⇒ 결과 객체에는 다음 값인 value와 다음 반환 값이 더이상 없는 경우 나타나는 done 프로퍼티가 있음
⇒ 값 컬렉션 내의 위치에 대한 내부 포인터를 유지하고 next() 메서드를 호출할 때마다 다음 적절한 값을 반환
